grammar.js

```diff
@@ -451,12 +451,9 @@ const rules = {
   function_type_specifier: $ =>
     seq(
       rep($._type_modifier),
-      '(',
-      'function',
-      '(',
+      '(function(',
       com.opt(opt($.inout_modifier), $._type, opt($.variadic_modifier), ','),
-      ')',
-      ':',
+      '):',
```

Comment:
Unfortunately, concatenating the function_type_specifier tokens this way prevents some valid Hack from being parsed. Having spaces in between all these tokens is valid Hack that the generated parser won't be able to parse.

E.g. Hack can parse this, but the updated grammar can't
```
(( function () : void) $test): void ==> {};
```

Reply:
No ideas come to mind for a solution right away, but can mess around with this later and see if I have any suggestions. Curious to hear what you tried.

Reply:
I think regex whitespace matching will work. Let me try and see what happens.